core/src/main/scala/io/qbeast/core/transform/LinearTransformer.scala

```diff
@@ -26,18 +37,60 @@ case class LinearTransformer(columnName: String, dataType: QDataType) extends Tr
     }
   }
 
+  private def generateRandomNumber(min: Any, max: Any): Any = {
+    val random = Random.nextDouble()
+
+    dataType match {
+      case DoubleDataType =>
+        min
+          .asInstanceOf[Double] + (random * (max.asInstanceOf[Double] - min.asInstanceOf[Double]))
+      case IntegerDataType =>
+        min.asInstanceOf[Int] + (random * (max.asInstanceOf[Int] - min.asInstanceOf[Int])).toInt
+      case LongDataType =>
+        min.asInstanceOf[Long] + (random * (max.asInstanceOf[Long] - min
+          .asInstanceOf[Long])).toLong
+      case FloatDataType =>
+        min.asInstanceOf[Float] + (random * (max.asInstanceOf[Float] - min
+          .asInstanceOf[Float])).toFloat
+      case DecimalDataType =>
+        min
+          .asInstanceOf[Double] + (random * (max.asInstanceOf[Double] - min.asInstanceOf[Double]))
+    }
+  }
+
+  private lazy val (minValue, maxValue): (Any, Any) = {
+    dataType match {
+      case DoubleDataType => (Double.MinPositiveValue, Double.MaxValue)
+      case IntegerDataType => (Int.MinValue, Int.MaxValue)
+      case LongDataType => (Long.MinValue, Long.MaxValue)
+      case FloatDataType => (Float.MinValue, Float.MaxValue)
+      case DecimalDataType => (Double.MinValue, Double.MaxValue)
+    }
+  }
+
   override def stats: ColumnStats =
     ColumnStats(
       statsNames = Seq(colMax, colMin),
       statsSqlPredicates = Seq(s"max($columnName) AS $colMax", s"min($columnName) AS $colMin"))
 
   override def makeTransformation(row: String => Any): Transformation = {
-    val min = getValue(row(colMin))
-    val max = getValue(row(colMax))
-    assert(min != null && max != null)
+    val minAux = row(colMin)
+    val maxAux = row(colMax)
+    val (min, max) =
+      // TODO If all values are null...
```

Comment:
Hey! Opening a discussion here: what should be the measure when **all values** are null? 
Right now,  min and max are initialized with the minimum and maximum values of the dataType.

Reply:
The ideal is that we remove this dimension, and we add it later when we find no null data we issue a new revision with the new dimension. In the end, we should distinguish between the columns that the user thinks we should take into consideration and the ones that are really necessary/useful. I think is best if we disconnect the two, so we can simply drop columns where all data is null or have the same value

Reply:
I opened a PR on my fork to check if it makes sense to drop columns that have the same value (null or not)
https://github.com/osopardo1/qbeast-spark/pull/1

Please check it when you can @eavilaes @cugni ! :pray: Thank you!

Reply:
Update: If all values are null, we pick null/zero value. PR osopardo1#1 is left for later